keanu-project/src/main/java/io/improbable/keanu/vertices/Vertex.java

```diff
@@ -249,4 +305,32 @@ public String toString() {
         }
         return stringBuilder.toString();
     }
+
+    public final KeanuSavedBayesNet.Vertex toProtoBuf() {
+        KeanuSavedBayesNet.Vertex.Builder vertexBuilder = KeanuSavedBayesNet.Vertex.newBuilder();
+
+        if (label != null) {
+            vertexBuilder = vertexBuilder.setLabel(label.toString());
+        }
+
+        if (isConstantVertex(this)) {
```

Comment:
The fact that you need to check the type here feels like a code smell. Can you instead put the `toProtoBuf` logic into a separate class, and use polymorphism (maybe the visitor pattern) to decide whether to `setConstantValue`?

Reply:
Agreed - it was a small quick addition, but I didn't want to overcomplicate the API too much for what each Vertex.  Interested if you have an example of how you'd like to see this work.